extensions/flush-dns/src/index.ts

```diff
@@ -1,66 +1,134 @@
 import { confirmAlert, showHUD } from "@raycast/api";
-import { SpawnSyncReturns, execSync } from "node:child_process";
+import { execSync } from "node:child_process";
+import { platform } from "node:os";
 
-const isSpawnReturn = (e: unknown): e is SpawnSyncReturns<Buffer> => {
-  return typeof e === "object" && e !== null && "status" in e && "stdout" in e && "stderr" in e;
-};
+type ShellType = string | undefined;
+
+interface CommandSet {
+  commands: Record<string, string>;
+  shell: ShellType;
+}
 
-const commands = {
-  mDNSResponder: "sudo /usr/bin/killall -HUP mDNSResponder",
-  dscacheutil: "sudo /usr/bin/dscacheutil -flushcache",
-  mdnsflushcache: "sudo /usr/bin/discoveryutil mdnsflushcache",
-} as const satisfies Record<string, string>;
+const OS_COMMANDS: Record<string, CommandSet> = {
+  darwin: {
+    commands: {
+      dscacheutil: "/usr/bin/dscacheutil -flushcache",
+      mDNSResponder: "/usr/bin/killall -HUP mDNSResponder",
+      mdnsflushcache: "/usr/bin/discoveryutil mdnsflushcache",
+    },
+    shell: "/bin/bash",
+  },
+  win32: {
+    commands: {
+      ipconfig: "ipconfig /flushdns",
+    },
+    shell: undefined, // Uses cmd.exe by default
```

Comment:
Should this be undefined or `"cmd.exe"` in that case. Feels like cmd.exe is making more sense, but I don't have a strong opinion regarding this.

Reply:
raycast windows don't have showHUD api right so hard to know if throwing error, but Nodejs docs says right [here ](https://nodejs.org/api/child_process.html#child_processexecsynccommand-options) node uses process.env.ComSpec as default shell as cmd.exe